src/hooks/transactions/useEstimateFee.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges UI.
+//
+// Parity Bridges UI is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Parity Bridges UI is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges UI.  If not, see <http://www.gnu.org/licenses/>.
+
+import { useCallback } from 'react';
+import { useApiCallsContext } from '../../contexts/ApiCallsContextProvider';
+import { useSourceTarget } from '../../contexts/SourceTargetContextProvider';
+import { TransactionActionCreators } from '../../actions/transactionActions';
+import { useUpdateTransactionContext, useTransactionContext } from '../../contexts/TransactionContext';
+import { getSubstrateDynamicNames } from '../../util/getSubstrateDynamicNames';
+import useLaneId from '../chain/useLaneId';
+
+export const useEstimateFee = () => {
+  const { createType, stateCall } = useApiCallsContext();
+  const laneId = useLaneId();
+  const {
+    sourceChainDetails: { chain: sourceChain },
+    targetChainDetails: { chain: targetChain }
+  } = useSourceTarget();
+  const { dispatchTransaction } = useUpdateTransactionContext();
+  const { estimatedFeeMethodName } = getSubstrateDynamicNames(targetChain);
+  const { payload } = useTransactionContext();
+
+  const getEstimateFeeCall = useCallback(async () => {
+    if (!payload) {
+      return;
+    }
```

Comment:
So seems that in case the payload ends up being empty we will still keep previously calculated fee in the state.

Reply:
Right, this problem has gone with the implementation of `useDispatchGenericCall`